### ISSUE TITLE
Sidenote: Annotate command not found error

### ISSUE THREAD

**Issue by dewongway** [cyberbiont/vscode-sidenotes#1]
When calling the Sidenote: Annotate command from the command pallete, it encountered the following error:

Command 'Sidenote: Annotat' resulted in an error (command 'sidenotes.annotate' not found).

Version: 1.60.1 (user setup)
Commit: 83bd43bc519d15e50c4272c6cf5c1479df196a4d
Date: 2021-09-10T17:07:10.714Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.18363

sidenote version: 1.0.0

**Comment by lundeen-bryan** [comment]
Same error. I tried it in the following languages:

- Visual Basic For Applications
- HTML
- Python

Got the same error in all three languages. 
I tried using the command pallete and selecting Annotate and go the same result.
I also tried using right click and selecting annotate (code) and same result. 

Says command 'sidenotes.annotate' not found. 

Hope this is just a temporary problem, but I do notice this issue was created back in 2021. 

Please comment if there is a new patch. Thank you.

**Comment by cyberbiont** [comment]
Hi, maybe a stupid question, but have you tried to reload your VS Code?
I rarely notice this problem when the extension fails to work for some reason, reload fixes it.

**Comment by lundeen-bryan** [comment]
> Hi, maybe a stupid question, but have you tried to reload your VS Code? I rarely notice this problem when the extension fails to work for some reason, reload fixes it.

It's a good question and I forgot to mention that I did try reloading. 
After realoading when my cursor does not have text selected and I use Annotate then I get an error, but if I have a comment selected, and use the Annotate command, it works. 

So this seems to be partially fixed. Do you know of any YouTube videos to show how it's used? I may be using it incorrectly.

**Comment by cyberbiont** [comment]
Sorry I cannot reproduce what you describe here, but I'll make a note.
I'm going to refactor this extension a bit and release a new version soon, so stay put, maybe this problem will go away.
I don't have any Youtube videos currently, but it's a good idea, maybe I should make one to illustrate different use cases.
I'll a link here in README in case if I'll manage to record it

**Comment by lundeen-bryan** [comment]
Thank you so much. I'm going to run some more tests. I'll test with no other VS Code extension, and I'll try on other computers. So at least 2 tests I can think of off top of my head. If it works under one of those circumstances, I'll start testing by adding other VS Code extensions added one-by-one. 

As I said before, GIVEN that I have text selected, AND I use F1 to select the Annotate command, THEN it will add the little book icon followed by my name and what looks like a CLSID number. Then when I look in my .vscode folder for that project, Then I see a matching ".md" file with that CLSID as the title with the selected text in the markdown document. 

From what I can see from your gifs, it looks like you are also seeing a panel open on the right side of the VS Code active window. I can only open that window manually by dragging the file to the right side of the VS Code window. 

To be honest even in the circumstances listed above, it's a very useful extension. Thank you so much. I starred the repo and I'll add a review on the Marketplace as soon as you update it. I don't want to put a review while you are still developing additional features. 
Anyway, thank you so much!

**Comment by prmichaelsen** [comment]
Note: Restarting VS code fixed the command not found error.

https://github.com/cyberbiont/vscode-sidenotes/assets/8428140/fbf269f4-d40f-4a69-a57e-28bc7ef2b528

However comment is not Cmd + Clickable to navigate to comment as I would anticipate. To open comment, I Ctrl + C the UUID and Ctrl + P to navigate to the file which is not a great workflow. Curious if that is a bug.


```

Version: 1.81.1 (Universal)
Commit: 6c3e3dba23e8fadc360aed75ce363ba185c49794
Date: 2023-08-09T22:20:33.924Z (3 wks ago)
Electron: 22.3.18
ElectronBuildId: 22689846
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Darwin x64 21.6.0
```

Note, similar libraries:

* [marginalia](https://marketplace.visualstudio.com/items?itemName=jamesnorton.marginalia)
* [linenote](https://marketplace.visualstudio.com/items?itemName=tkrkt.linenote) - uses line number, note locations do not update properly when lines are updated

**Comment by lundeen-bryan** [comment]
> However comment is not Cmd + Clickable to navigate to comment as I would anticipate. To open comment, I Ctrl + C the UUID and Ctrl + P to navigate to the file which is not a great workflow. Curious if that is a bug.

Not a bug. Here is how you should do it. Sorry I don't have a Mac so you'll have to translate to Mac keyboard shortcuts. But you essentially create the sidenote as you did in your gif, then if your cursor is in the UUID and you press "Alt+P" then that UUID will show up as a file that you are searching for. Then hold down Ctrl and press ENTER then it will open the note to the side. Make your notes, then close the sidenote. 

This is how I'm using it.

**Comment by cyberbiont** [comment]
> Note: Restarting VS code fixed the command not found error.
> Screen.Recording.2023-09-01.at.4.58.50.AM.mov
> 
> However comment is not Cmd + Clickable to navigate to comment as I would anticipate. To open comment, I Ctrl + C the UUID and Ctrl + P to navigate to the file which is not a great workflow. Curious if that is a bug.
> 
> ```
> 
> Version: 1.81.1 (Universal)
> Commit: 6c3e3dba23e8fadc360aed75ce363ba185c49794
> Date: 2023-08-09T22:20:33.924Z (3 wks ago)
> Electron: 22.3.18
> ElectronBuildId: 22689846
> Chromium: 108.0.5359.215
> Node.js: 16.17.1
> V8: 10.8.168.25-electron.0
> OS: Darwin x64 21.6.0
> ```
> 
> Note, similar libraries:
> 
>     * [marginalia](https://marketplace.visualstudio.com/items?itemName=jamesnorton.marginalia)
> 
>     * [linenote](https://marketplace.visualstudio.com/items?itemName=tkrkt.linenote) - uses line number, note locations do not update properly when lines are updated

Yeah, this error ('command not found') happens occasionally, when something goes wrong with extension initialization.
I still haven't found a time to debug it, but it's easily fixed, you can restart VS Code or execute 'Developer: Restart Extension Host'  command'

As for opening the notes: yeah it's not cmd (ctrl) + clilck. You can either:
- hover over the uuid, wait for tooltip to show up and press 'Edit' there (that's how I usually do it)

- like @lundeen-bryan mentioned (thanks for assistance), you can put your cursor on the uuid line and call 'annotate' command (you can assign whatever key combination you like to it), that's intended for keyboard people mostly.


Thanks for mentioning other extensions, I tried them (at least marginalia) but wasn't happy with the way they work that's why I wrote this one)

**Comment by lundeen-bryan** [comment]
I like this ext a lot, especially for working in teams. Once you are ready for production, you use the ext to clear all notes (at least I was told we don't leave our comments in production code, instead create documentation. So you can actually use your sidenotes (since they are markdown files) to create at least some starting documentation, then clear notes from the dev branch and push to the production branch (via your regular process).  Great ideas in this extension. Thank you!

**Comment by prmichaelsen** [comment]
I might open a PR to make them Cmd + Clickable. We'll see 😉

It's definitely feasible via a DocumentLinkProvider. Would that be welcome if I did so?

**Comment by prmichaelsen** [comment]
Loaded up the extension source and was playing with it. Turns out the issue I was experiencing appears to be with the regex not actually matching note markers. This explains why hover tooltips weren't showing either. I didn't spend any time debugging, but here is a link to the NoteLinkProvider that would be implemented: https://github.com/cyberbiont/vscode-sidenotes/compare/main...prmichaelsen:vscode-sidenotes:main

An approach like this worked in a similar project I'm working on. Unfortunately I had to hack a bit to get a hold of a `utils` instance. That said, I'm welcome to feedback, if someone figures out the regex problem, I'd be happy to finish the feature. Cheers 🍻

**Comment by lundeen-bryan** [comment]
> Turns out the issue I was experiencing appears to be with the regex

Can you expand on that? It sounds interesting that regex was the problem. 
First, you should document again the steps to reproduce your problem and start a new thread with a different target issue. This way, the dev and anyone who wants to help can see the problem in an isolated thread. If possible, write out the expected behavior of the app vs. the actual behavior. Write out the steps you are doing currently vs. the expected steps to have the behavior you want. I think doing this would help get the situation clarified.

**Comment by cyberbiont** [comment]
Hi guys, thanks for participation.
@prmichaelsen  sure, feel free to open a PR. I'll take a look once I have time. unfortunately I am still under the tight deadline currently on my main work and will be able to look into this only after a couple of weeks.

@lundeen-bryan I am really happy to hear that someone actually uses this as a collaboration tool
I added this feature as an experiment, glad that it turned out useful

**Comment by prmichaelsen** [comment]
The changes ended up being pretty comprehensive. The logic for using hashes ended up requiring restructuring most of the extension. I decided to fork the extension in the mean time. You can check out the repo at https://github.com/prmichaelsen/linenoteplus if you're interested 

Whoops! Mixed up my threads, I thought I was on a linenote issue.

I did open a PR for this, let me see if I can reopen it.
